mssql_script: add tds_version and encryption parameters

Halo1236 opened the pull request:
##### SUMMARY

This pull request adds support for specifying `tds_version` and `encryption` options when connecting to Microsoft SQL Server in the `mssql_script` Ansible module. These options improve compatibility with older SQL Server versions and allow more flexible connection configurations.

**Connection options enhancements:**

* Added two new module parameters, `tds_version` and `encryption`, to allow users to specify the TDS protocol version and encryption settings for SQL Server connections.
* Updated the connection logic to pass the `tds_version` and `encryption` parameters to `pymssql.connect`, enabling these options to be used when establishing the database connection. [[1]](diffhunk://#diff-d9b91513b25ca15554c221d60c719bfd30d08f13f9f0770d7b35bfabf658ec18R320-R322) [[2]](diffhunk://#diff-d9b91513b25ca15554c221d60c719bfd30d08f13f9f0770d7b35bfabf658ec18L329-R334)

##### ISSUE TYPE

- Bugfix Pull Reques


##### COMPONENT NAME

mssql_script

felixfontein commented:
Also, please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment). Thanks.

Halo1236 commented:
@felixfontein Thank you for your guidance! I have updated the code and the changelog. There is an issue in Issues that has the same error message as the one I encountered during my testing, but I am not sure if my update can resolve their problem. Therefore, I have removed the association for now.

felixfontein commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,14 @@
+  encryption:
+    description:
+      - Specify whether to use encryption for the connection to the server, Please refer to the pymssql documentation for detailed information.
+    type: str
```
```suggestion
    type: str
    version_added: 11.4.0
```

felixfontein commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,14 @@
+  tds_version:
+    description:
+      - Specify the TDS protocol version to use when connecting to the server, Please refer to the pymssql documentation for detailed information.
+    type: str
```
```suggestion
    type: str
    version_added: 11.4.0
```

felixfontein commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,14 @@
     version_added: 8.4.0
+  encryption:
+    description:
+      - Specify whether to use encryption for the connection to the server, Please refer to the pymssql documentation for detailed information.
```
```suggestion
      - Specify whether to use encryption for the connection to the server.
        Please refer to the pymssql documentation for detailed information.
```

felixfontein commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,14 @@
+    type: str
+  tds_version:
+    description:
+      - Specify the TDS protocol version to use when connecting to the server, Please refer to the pymssql documentation for detailed information.
```
```suggestion
      - Specify the TDS protocol version to use when connecting to the server.
        Please refer to the pymssql documentation for detailed information.
```

russoz commented on plugins/modules/mssql_script.py:
```diff
@@ -291,6 +303,8 @@ def run_module():
         output=dict(default='default', choices=['dict', 'default']),
         params=dict(type='dict'),
         transaction=dict(type='bool', default=False),
+        tds_version=dict(type='str', required=False),
+        encryption=dict(type='str', required=False)
```
`required=False` is redundant, please remove

russoz commented on plugins/modules/mssql_script.py:
```diff
@@ -325,8 +342,17 @@ def run_module():
     try:
-        conn = pymssql.connect(
-            user=login_user, password=login_password, host=login_querystring, database=db)
+        kwargs = {
```
Nit picking => maybe call this variable `connection_args`?

russoz commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,18 @@
+        Please refer to the pymssql documentation for detailed information.
+    type: str
+    version_added: 11.4.0
+  tds_version:
```
It would be nice to add one or two examples in the documentation.

russoz reviewed (commented):
Hi @Halo1236 thanks for your contribution!

I've left a couple of comments, nothing major. Other than that, it LGTM

felixfontein commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,20 @@
+    description:
+      - Specify whether to use encryption for the connection to the server.
+        Please refer to the pymssql documentation for detailed information.
+      - The available values are "off" and "on"
```
```suggestion
      - The available values are V("off") and V("on").
```
(I left the quotes to avoid conversion to booleans.)

If no other values are allowed, how about adding `choices` to limit the input to these two values?

russoz commented on plugins/modules/mssql_script.py:
```diff
@@ -62,6 +62,20 @@
+        Please refer to the pymssql documentation for detailed information.
+      - The available values are "off" and "on"
+    type: str
+    version_added: 11.4.0
```
This is now:
```suggestion
    version_added: 12.0.0
```

russoz commented on plugins/modules/mssql_script.py:
```diff
@@ -325,8 +356,17 @@ def run_module():
+            "database": db,
+        }
+        if encryption is not None:
+            connection_args["encryption"] = encryption
```
if you make the parameter a boolean, then this should go:
```suggestion
            connection_args["encryption"] = "on" if encryption else "off"
```

russoz reviewed (commented):
Hi @Halo1236 

A couple of additional comments. The PR is looking good, it just need these few adjustments.

If you are struggling with the tests, may I suggest giving a try to `andebox`, by yours truly:

https://pypi.org/project/andebox/

You can run this sanity test easily in your local machine by simply running:

```
$ andebox test -- sanity --docker default --python 3.13 plugins/modules/your_module.py
```

felixfontein commented:
Please note that this PR now has a conflict since we reformatted all code in the collection (see #10999).

felixfontein commented:
@Halo1236 ping

needs_info
